cpu/cpu.go

```diff
@@ -146,12 +146,20 @@ func (p *Plugin) GetMetricTypes(cfg plugin.ConfigType) ([]plugin.MetricType, err
 // It returns error in case retrieval was not successful
 func (p *Plugin) CollectMetrics(metricTypes []plugin.MetricType) ([]plugin.MetricType, error) {
 	metrics := []plugin.MetricType{}
-	for _, metricType := range metricTypes {
-		if !p.initialized {
-			if err := p.init(metricType.Config().Table()); err != nil {
-				return nil, err
-			}
+	if len(metricTypes) == 0 {
+		return nil, nil
+	}
+	if !p.initialized {
+		if err := p.init(metricTypes[0].Config().Table()); err != nil {
+			return nil, err
 		}
+	} else {
```

Comment:
Not sure I understand your point here, the purpose of the else is the very core of the fix for this issue: getStats not being called in subsequent calls of metric gathering. Please advise